tests/components/hisense_aehw4a1/test_init.py

```diff
@@ -0,0 +1,73 @@
+"""Tests for the Hisense AEH-W4A1 init file."""
+from unittest.mock import patch
+
+from homeassistant import config_entries, data_entry_flow
+from homeassistant.components import hisense_aehw4a1
+from homeassistant.setup import async_setup_component
+
+from tests.common import mock_coro
+
+
+async def test_creating_entry_sets_up_climate_discovery(hass):
+    """Test setting up hisense_aehw4a1 loads the climate component."""
+    with patch(
+        "homeassistant.components.hisense_aehw4a1.climate.async_setup_entry",
+        return_value=mock_coro(True),
+    ) as mock_setup:
+        result = await hass.config_entries.flow.async_init(
+            hisense_aehw4a1.DOMAIN, context={"source": config_entries.SOURCE_DISCOVERY}
```

Comment:
This shouldn't be source discovery I think since that will never happen, right? It will be the user who triggers the config flow.

Reply:
Right